syft/cataloger/python/parse_setup.go

```diff
@@ -0,0 +1,46 @@
+package python
+
+import (
+	"bufio"
+	"io"
+	"regexp"
+	"strings"
+
+	"github.com/anchore/syft/syft/cataloger/common"
+	"github.com/anchore/syft/syft/pkg"
+)
+
+// integrity check
+var _ common.ParserFn = parseSetup
+
+var pinnedDependency = regexp.MustCompile(`['"]\W?(\w+\W?==\W?[\w\.]*)`)
```

Comment:
nit: there is an opportunity to capture more environment marker cases with `['"]\s*([\w]+\s*==\s*[\w\.]*)\s*[;'"]` which would find `ipaddress==1.0` in:
```
setup(
    ...
    install_requires=[
        'six',
        'humanize',
    ],
    extras_require={
        ':python_version == "2.7"': [
            'ipaddress==1.0',
        ],
    },
)
``` 
Not necessary to incorporate in this PR.